auth/couchdb_session_authenticator.go

```diff
@@ -221,8 +226,13 @@ func (a *CouchDbSessionAuthenticator) requestSession() (*session, error) {
 			Headers:    resp.Header,
 			RawResult:  buff.Bytes(),
 		}
-		err := fmt.Errorf(buff.String())
-		return nil, core.NewAuthenticationError(detailedResponse, err)
+
+		err := &core.HTTPProblem{
+			IBMProblem: core.IBMErrorf(nil, core.NewProblemComponent("", ""), buff.String(), ""),
```

Comment:
a little unsure on this empty component

Reply:
Yes, this one is awkward. Core [deprecated `NewAuthenticationError`](https://github.com/IBM/go-sdk-core/blob/main/core/authentication_error.go#L28) which is a pity, it was one handy util function.

The recommended `HTTPProblem` doesn't have initializer, so I lifted [code from `httpErrorf`](https://github.com/IBM/go-sdk-core/blob/main/core/authentication_error.go#L48) and the empty component comes from [the function itself](https://github.com/IBM/go-sdk-core/blob/main/core/http_problem.go#L140)

It has `buff.String()` as a summary which is our error message, so it's not entirely empty.

Theoretically we can also run it through [`EnrichHTTPProblem`](https://github.com/IBM/go-sdk-core/blob/main/core/http_problem.go#L149), but then we'd have to define `operationID` and `ProblemComponent` for it at which moment I kind of gave up.

Reply:
I thought about it a little bit more and I think you right, we can do better than empty struct here, even if that's more verbose. Updated in https://github.com/IBM/cloudant-go-sdk/pull/474/commits/fa5d41a6dafe11c883893d7f5b2c0199703d8c1c

Reply:
Thanks. I wonder if we should ask for a `HTTPProblem` initializer - I was trying to think if there were any use cases beyond auth that might make it more widely applicable.